robot_skills/src/robot_skills/robot.py

```diff
@@ -126,34 +126,75 @@ def tf_transform_pose(self, ps, frame):
         output_pose = self.tf_listener.transformPose(frame, ps)
         return output_pose
 
-    def get_arm(self, side):
-        """Get an arm object and a backup for that arm by giving a side as either a string or an Arm-object
-        @param side Either string from robot.arms.keys() or Arm from robot.arms.values()
-        >>> robot = Robot("dummy")
-        >>> arm, backup_arm = robot.get_arm("left")
-        >>> assert(arm == robot.leftArm)
-        >>> assert(backup_arm == robot.rightArm)"""
-        preferred_side = self.arms[self.arms.keys()[0]]
-
-        #Define which arm is which's backup arm (left backs up for right etc)
-        backup_arms = self.arms.values() #Get a *list* of arms i.e. the values of the arm-dict, not the keys
-        backup_arms.insert(0, backup_arms.pop()) #Make the last arm the first in the list, so we shift by 1
-        backup_str_dict = dict(zip(self.arms.keys(), backup_arms)) #Create a dict again that maps strings to backup-arms
-        backup_obj_dict = {self.arms[side]:backup_str_dict[side] for side in self.arms.keys()} #Create a dict that maps e.g. self.LeftArm to self.rightArm
-
-        if isinstance(side, basestring):
-            try:
-                preferred_side = self.arms[side]
-            except KeyError:
-                print "Unknown arm side:" + str(side) + ". Defaulting to 'right'"
-                preferred_side = self.arms[self.arms.keys()[0]]
-        elif isinstance(side, arms.Arm):
-            preferred_side = side
-        else:
-            print "Unknown arm side:" + str(side) + ". Defaulting to '{0}'".format(preferred_side.side)
+    def get_robot_arm(self, gripper_type=None, goals=None, trajectories=None, preferred=None):
```

Comment:
Add parameter to get an arm that is occupied by a specific Ed Entity. 
This trumps other parameters, I think. It makes no sense to specify both `required` and `occupied_by`. 

Also, when specifying `occupied_by=None`, does that mean we don't care or that we want an unoccupied arm? 
I think that `occupied_by=False` 

Or `set()` for unoccupied, and `set("<some id>")` for a specific item and `None` for 'don't care', or this but with lists: `[], ["some id"], None`